core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -641,12 +608,12 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId, boo
     return loc;
   }
 
-  @Nullable
-  private StorageDirView allocateSpace(long sessionId, AllocateOptions options) {
-    StorageDirView dirView = null;
+  private StorageDirView allocateSpace(long sessionId, AllocateOptions options)
+      throws WorkerOutOfSpaceException, IOException {
+    StorageDirView dirView;
     BlockMetadataView allocatorView =
         new BlockMetadataAllocatorView(mMetaManager, options.canUseReservedSpace());
-    try {
+    while (true) {
```

Comment:
When does this loop run the 2nd time?

Reply:
it will never loop again. we will break or return from the 1st loop

Reply:
Yea, I guess it makes sense to add a comment before this loop mentioning it will never run again, and the only intention is to handle the undesired cases (where we `break` from the loop) together by throwing an exception. What do you think?

Reply:
make sense. fixed